3D-microservices-observability/node-js-app/beachshirt/controllers/deliveryController.js

```diff
@@ -0,0 +1,47 @@
+/**
+* Driver for delivery service which is responsible to deliver the shirts of a given style.
+* @author Yogesh Prasad Kurmi (ykurmi@vmware.com)
+*/
+const uuidv1 = require('uuid/v1');
+
+module.exports = (app, log) => {
+
+    app.post('/dispatch/:orderNum/', (req, res) => {
+        let orderNum = req.params.orderNum;
+        let shirts = JSON.parse(req.body.shirts);
+        if (Math.floor(Math.random() * 5) == 0) {
+            let msg = "Failed to order shirts!";
+            log.error(msg);
+            return res.status(503).json({ error: msg });
+        }
+
+        if (Math.floor(Math.random() * 10) == 0) {
+            let msg = "Invalid Order Num";
+            log.error(msg);
+            return res.status(400).json({ error: msg });
+        }
+
+        if (Math.floor(Math.random() * 10) == 0) {
+            shirts = null;
+        }
+        if (shirts == null || shirts.length == 0){
+            let msg = "No shirts to deliver";
+            log.info(msg);
+            return res.status(400).json({ error: msg });
```

Comment:
Minor points, but you can define msg as error and use const instead of let (L28-30):
`
const error = "No shirts to deliver";
log.info(error);
return res.status(400).json({ error });
`

This is also true for L13-15 and L19-21